doc/west/repo-tool.rst

```diff
@@ -0,0 +1,270 @@
+.. _west-multi-repo:
+
+Multi-repository management
+###########################
+
+.. note::
+
+   Zephyr is currently not managed as a multi-repo. This page describes an
+   upcoming tool and potential workflow.
+
+West includes a set of commands for working with projects composed of multiple
+Git repositories (a *multi-repo*), similar to Google's `repo
+<https://gerrit.googlesource.com/git-repo/>`_ tool.
+
+The rest of this page introduces multi-repo concepts and gives an overview of
+the multi-repo commands, along with an example workflow.
+
+.. note::
+
+   The multi-repo commands are meant to augment Git in minor ways for
+   multi-repo work, not replace it. For tasks that aren't multi-repo-related,
+   use plain Git commands.
+
+   This page explains what the West multi-repo commands do behind the scenes.
+
```

Comment:
I'm missing a paragraph here explaining how `west` and `manifest` are special repositories also handled by `west`, and how that works with the rest. It's sort of connected to the multi-repo itself. Maybe what you've added below is enough, but I am not sure, especially for users that are less savvy.

Reply:
I added a note that mentions how the self-update for West itself is implemented ("To implement self-updates, west init also clones...").

IMO, the manifest update is the most important thing. The West self-update is more of an internal detail.